lib/NagiosStatusConnector.php

```diff
@@ -30,55 +34,43 @@ public function __construct()
     {
         parent::__construct();
 
-        $this->url = $this->configuration->getString(self::NAGIOS_URL, '');
-        $this->certPath = $this->configuration->getString(self::NAGIOS_CERT_PATH, '');
-        $this->certPassword = $this->configuration->getString(self::NAGIOS_CERT_PASSWORD, '');
-        $this->caPath = $this->configuration->getString(self::NAGIOS_CA_PATH, '');
-        $this->peerVerification = $this->configuration->getBoolean(self::NAGIOS_PEER_VERIFY, false);
-
-        if (empty($this->url)) {
-            throw new \Exception('Required option \'' . self::NAGIOS_URL . '\' is empty!');
-        } elseif (empty($this->certPath)) {
-            throw new \Exception('Required option \'' . self::NAGIOS_CERT_PATH . '\' is empty!');
-        } elseif (empty($this->caPath)) {
-            throw new \Exception('Required option \'' . self::NAGIOS_CA_PATH . '\' is empty!');
+        $this->params = $this->configuration->getArray(self::STATUS_NAGIOS, []);
+
+        if (empty($this->params[self::HOST])) {
+            throw new Exception('Required option \'' . self::HOST . '\' is empty!');
+        } elseif (empty($this->params[self::KEY_PATH])) {
+            throw new Exception('Required option \'' . self::KEY_PATH . '\' is empty!');
+        } elseif (empty($this->params[self::LOGIN])) {
+            throw new Exception('Required option \'' . self::LOGIN . '\' is empty!');
+        } elseif (empty($this->params[self::COMMAND])) {
+            throw new Exception('Required option \'' . self::COMMAND . '\' is empty!');
         }
+
+        $this->host = $this->params[self::HOST];
+        $this->keyPath = $this->params[self::KEY_PATH];
+        $this->login = $this->params[self::LOGIN];
+        $this->command = $this->params[self::COMMAND];
     }
 
 
     public function getStatus()
     {
         $result = [];
-        $serviceStatuses = [];
 
-        $ch = curl_init();
-        curl_setopt($ch, CURLOPT_URL, $this->url);
-        curl_setopt($ch, CURLOPT_VERBOSE, true);
-        curl_setopt($ch, CURLOPT_SSL_VERIFYPEER, $this->peerVerification);
-        curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
-        curl_setopt($ch, CURLOPT_SSLCERT, $this->certPath);
-        curl_setopt($ch, CURLOPT_CAPATH, $this->caPath);
-        curl_setopt($ch, CURLOPT_SSLKEYPASSWD, $this->certPassword);
+        $key = RSA::load(file_get_contents($this->keyPath));
```

Comment:
I think we should check for two cases here
- According to the documentation, if the _file_get_contents_ will fail, it will return false. I think this case should be handled by giving a proper explanation of what has happened.
- Another thing is that the _file\_get\_contents_ cannot read special characters in the path and these have to be encoded using the _urlencode_. Just to be sure I would add this wrapper on the key path property.

Reply:
I added check if file_get_contents returns false and throw exception in this case